src/generate.ts

```diff
@@ -8,28 +8,37 @@
 
 import * as fs from 'fs'
 import deepEqual from 'deep-equal'
-import { Route, ResolvedOptions } from './types'
+import { Route, ResolvedOptions, PageDirOptions } from './types'
 import { debug, isDynamicRoute } from './utils'
 import { stringifyRoutes } from './stringify'
 import { tryParseCustomBlock, parseSFC } from './parseSfc'
 
 function prepareRoutes(
   routes: Route[],
   options: ResolvedOptions,
+  pagesDirOptions: PageDirOptions,
   parent?: Route,
 ) {
   for (const route of routes) {
-    if (route.name)
+    if (route.name) {
       route.name = route.name.replace(/-index$/, '')
+      if (pagesDirOptions.baseRoute)
+        route.name = `${pagesDirOptions.baseRoute}-${route.name}`
+    }
 
-    if (parent)
+    if (parent) {
       route.path = route.path.replace(/^\//, '').replace(/\?$/, '')
+    }
+    else {
+      if (pagesDirOptions.baseRoute)
+        route.path = `/${pagesDirOptions.baseRoute}${route.path}`
```

Comment:
```suggestion
        const baseRoute = `/${pagesDirOptions.baseRoute}`
        route.path === '/' ? baseRoute : baseRoute + route.path
```

Prevent something end with `/` in route path.